manual/src/tutorials/polymorphism.etex

```diff
@@ -280,30 +280,32 @@ With a regular polymorphic algebraic data type, the type parameters of
 the type constructor are constant within the definition of the type. For
 instance, we can look at arbitrarily nested list defined as:
 \begin{caml_example}{toplevel}
-  type 'a regular_nested = List of 'a list | Nested of 'a regular_nested list
-  let l = Nested[ List [1]; Nested [List[2;3]]; Nested[Nested[]] ];;
+type 'a regular_nested = List of 'a list | Nested of 'a regular_nested list;;
+let l = Nested[ List [1]; Nested [List[2;3]]; Nested[Nested[]] ];;
 \end{caml_example}
 Note that the type constructor "regular_nested" always appears as
 "'a regular_nested" in the definition above, with the same parameter
-"'a". Equipped with this type, one can compute a maximal depth with
+"'a". Equipped with this type, one can compute a depth with
 a classic recursive function
 \begin{caml_example}{toplevel}
-  let rec maximal_depth = function
+let rec regular_depth = function
   | List _ -> 1
-  | Nested [] -> 0
```

Comment:
I think that defining the depth of `Nested []` as `1` is a simpler specification. Like this we have 
`regular_depth (Nested [ List [] ])` = `regular_depth (Nested [ ])` and `depth (Nested _) > depth (List _)`.